src/NuGet.Core/NuGet.Protocol.Core.v3/Utility/LocalFolderUtility.cs

```diff
@@ -309,9 +309,9 @@ public static LocalPackageInfo GetPackagesConfigFolderPackage(string root, Packa
             // Try matching the exact version format
             var idVersion = $"{identity.Id}.{identity.Version.ToString()}";
             var expectedPath = Path.Combine(rootDirInfo.FullName, idVersion, $"{idVersion}{PackagingCoreConstants.NupkgExtension}");
-            var expectedFile = new FileInfo(expectedPath);
+            var expectedFile = TryCreateFileInfo(expectedPath);
 
-            if (expectedFile.Exists)
+            if (expectedFile != null && expectedFile.Exists)
```

Comment:
The actual error has to be logged at least at DEBUG level to troubleshoot the issue